packages/state-transition/src/cache/epochCache.ts

```diff
@@ -571,6 +606,15 @@ export class EpochCache {
     // ```
     this.epoch = computeEpochAtSlot(state.slot);
     this.syncPeriod = computeSyncPeriodAtEpoch(this.epoch);
+    // 6110 Only: Add current cpState.validators.length
+    // Only keep validatorLength for epochs after finalized cpState.epoch
+    // eg. [100(epoch 1), 102(epoch 2)].push(104(epoch 3)), this.epoch = 3, finalized cp epoch = 1
+    // We keep the last (3 - 1) items = [102, 104]
+    if (currEpoch >= this.config.EIP6110_FORK_EPOCH) {
+      this.historicalValidatorLengths = this.historicalValidatorLengths
+        .push(state.validators.length)
+        .slice((this.epoch - state.finalizedCheckpoint.epoch) * -1);
```

Comment:
You only need to slice if the finalize epoch advances, is the slice operation free if no change? Also please just reverse the subtraction, not sure how `*-1` helps on readability

Reply:
> You only need to slice if the finalize epoch advances, is the slice operation free if no change?

Yes, there are early return statement if deemed unnecessary to slice:
https://github.com/immutable-js/immutable-js/blob/1169ee1b57156737e058666d744ffca7c37eaa03/src/List.js#L178
and
https://github.com/immutable-js/immutable-js/blob/1169ee1b57156737e058666d744ffca7c37eaa03/src/List.js#L543